clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -613,15 +614,44 @@ public void testOffsetFetchRequestErroredRequests(final Errors error, final bool
         }
     }
 
+    @ParameterizedTest
+    @MethodSource("offsetFetchExceptionSupplier")
+    public void testOffsetFetchRequestTimeoutRequests(final Errors error,
+                                                      final boolean isRetriable) {
+        CommitRequestManager commitRequestManager = create(true, 100);
+        when(coordinatorRequestManager.coordinator()).thenReturn(Optional.of(mockedNode));
+
+        Set<TopicPartition> partitions = new HashSet<>();
+        partitions.add(new TopicPartition("t1", 0));
+        List<CompletableFuture<Map<TopicPartition, OffsetAndMetadata>>> futures = sendAndVerifyDuplicatedOffsetFetchRequests(
+                commitRequestManager,
+                partitions,
+                1,
+                error);
+
+        if (isRetriable) {
+            futures.forEach(f -> assertFalse(f.isDone()));
+
+            // Insert a long enough sleep to force a timeout of the operation. Invoke poll() again so that each
+            // OffsetFetchRequestState is evaluated via isExpired().
+            time.sleep(defaultApiTimeoutMs);
+            commitRequestManager.poll(time.milliseconds());
+            futures.forEach(f -> assertFutureThrows(f, TimeoutException.class));
+        } else {
+            futures.forEach(f -> assertFutureThrows(f, KafkaException.class));
+            assertEmptyPendingRequests(commitRequestManager);
```

Comment:
I guess we should be able to assert this no matter if retriable or not (just because in this test we're forcing a timeout on retriables).

Reply:
Actually this makes me wonder, are we removing the inflight fetch requests when we expire the request? I can only see them removed when we receive a response, but not when we intentionally expire the requests (where we remove only the unsent)

Reply:
I wasn't removing the inflight requests since the code in the response handler removes it. If we did remove the inflight requests before the response was received, the user would see the warning about the request not being in the "buffer" when it did complete.

Reply:
I see, still even if we leave the inflight, it would be helpful to assert that that we're not leaving any unsent after we expire them on ln 637/638. Something like this would be reassuring:
```
            assertFalse(commitRequestManager.pendingRequests.unsentOffsetFetches.isEmpty());
            commitRequestManager.poll(time.milliseconds());
            futures.forEach(f -> assertFutureThrows(f, TimeoutException.class));
            assertTrue(commitRequestManager.pendingRequests.unsentOffsetFetches.isEmpty());
```

Reply:
Added checks.